Yosemite/Yosemite/Stores/ShippingLabelStore.swift

```diff
@@ -118,9 +123,32 @@ private extension ShippingLabelStore {
         remote.packagesDetails(siteID: siteID, completion: completion)
     }
 
-    func checkCreationEligibility(siteID: Int64, orderID: Int64, isFeatureFlagEnabled: Bool, onCompletion: @escaping (_ isEligible: Bool) -> Void) {
-        // TODO-2971: implement shipping label creation eligibility check, hopefully with the new `/creation_eligibility` endpoint.
-        onCompletion(isFeatureFlagEnabled)
+    func checkCreationEligibility(siteID: Int64,
+                                  orderID: Int64,
+                                  canCreatePaymentMethod: Bool,
+                                  canCreateCustomsForm: Bool,
+                                  canCreatePackage: Bool,
+                                  onCompletion: @escaping (_ isEligible: Bool) -> Void) {
+        remote.checkCreationEligibility(siteID: siteID,
+                                        orderID: orderID,
+                                        canCreatePaymentMethod: canCreatePaymentMethod,
+                                        canCreateCustomsForm: canCreateCustomsForm,
+                                        canCreatePackage: canCreatePackage) { result in
+            switch result {
+            case .success(let eligibility):
+                if !eligibility.isEligible {
+                    if let reason = eligibility.reason {
+                        DDLogError("Order \(orderID) not eligible for shipping label creation: \(reason)")
+                    } else {
+                        DDLogError("Order \(orderID) not eligible for shipping label creation")
+                    }
+                }
```

Comment:
Does this seem reasonable or will this create too much noise in the app logs? I thought it might help in case users reach to support wondering why they can't create shipping labels for some or all of their orders, but it will generate quite a few logs if they aren't eligible.

Reply:
I think it makes sense to add these logs 👍